src/realm.js

```diff
@@ -135,6 +135,55 @@ export class Effects {
   }
 }
 
+export class LookasideTable {
```

Comment:
We need a better name. It's very similar to effects, so what distinguishes it from effects? (As a guidance for naming.) Is it an `EffectsBuilder`? Would be nice to distinguish mutable from immutable data structures.

Reply:
I guess this is like a builder that can take projections of multiple `Effects` and provide an immutable view of them to the `Serializer`. It's not quite an `Effects` builder because you can't apply the effects, and it encodes many possible program states. 

`FinalEffectsViewBuilder`? Not sure what name best represents its functionality.

Reply:
Another option: `PartialHeapSnapshotBuilder`.